File: DatabaseMigrationSystem.UseCases/Settings/Commands/SetSettingsCommand.cs
using DatabaseMigrationSystem.Common.Enums;
using MediatR;

namespace DatabaseMigrationSystem.UseCases.Settings.Commands;

/// <summary>
/// Установить настройки
/// </summary>
public class SetSettingsCommand : IRequest
{
    /// <summary>
    /// Строка подключения к исходной базе данных
    /// </summary>
    public string SourceConnectionString  { get; set; }
    
    /// <summary>
    /// Тип базы данных
    /// </summary>
    public DatabaseType SourceDatabaseType { get; set; }
    
    /// <summary>
    /// Строка подключения к базе данных
    /// </summary>
    public string DestinationConnectionString  { get; set; }
    
    /// <summary>
    /// Тип базы данных
    /// </summary>
    public DatabaseType DestinationDatabaseType { get; set; }
}

File: front/Database.Migration.System/src/app/api/models/set-settings-command.ts
/* tslint:disable */
/* eslint-disable */
import { DatabaseType } from './database-type';
export interface SetSettingsCommand {
  destinationConnectionString?: null | string;
  destinationDatabaseType?: DatabaseType;
  sourceConnectionString?: null | string;
  sourceDatabaseType?: DatabaseType;
}
